.github/workflows/release.yml

```diff
@@ -54,36 +54,60 @@ jobs:
     needs: [build-sdist]
     runs-on: ubuntu-latest
     container:
-      image: "ghcr.io/osgeo/gdal:ubuntu-small-3.8.5"
+      image: "ghcr.io/osgeo/gdal:ubuntu-small-3.9.2"
 
     steps:
+      - name: Install packages
+        run: |
+          apt-get update && apt-get install -y build-essential python3-dev
+
+      - name: Create virtual environment
+        # install uv and use it to create a virtual environment, then add it to
+        # environment variables so that it is automatically activated and can be
+        # used for tests below
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          . $HOME/.cargo/env
+          uv venv .venv
+          echo "VIRTUAL_ENV=.venv" >> $GITHUB_ENV
+          echo "$PWD/.venv/bin" >> $GITHUB_PATH
+
       - name: Download sdist from artifacts
         uses: actions/download-artifact@v4
         with:
           name: pyogrio-sdist
           path: wheelhouse
 
-      - name: Install pip
-        run: |
-          apt-get update && apt-get install -y python3-pip
-
       - name: Build from sdist and install test dependencies
         shell: bash
         run: |
-          python3 -m pip install --no-cache-dir wheelhouse/*.tar.gz
-          python3 -m pip install --no-cache-dir pytest pandas pyproj shapely>=2
-          python3 -m pip install --no-cache-dir --no-deps geopandas
-          python3 -m pip list
+          uv pip install --no-cache wheelhouse/*.tar.gz
+          uv pip install --no-cache pytest pandas pyproj shapely>=2
+          uv pip install --no-cache --no-deps geopandas
+          uv pip list
 
       - name: Run tests
         shell: bash
+        # virtual environment is automatically activated
         run: |
           cd ..
-          python3 -m pytest --pyargs pyogrio.tests -v
+          python -c "import pyogrio; print(f'GDAL version: {pyogrio.__gdal_version__}\nGEOS version: {pyogrio.__gdal_geos_version__}')"
```

Comment:
Not too important (this way is fine as well), but I _think_ the `uv run python -m pytest ..` should work here, because we are not in the root directory with a pyproject.toml file because of the `cd ..` (in contrast to the docker tests, I suppose)

Reply:
This worked fine, but we still need to set the environment variables to automatically activate `uv`, so it doesn't obviate the need for those.

Reply:
Not that it matters too much, but just because I wanted to try to understand `uv` better (new to it), I _think_ what is going on:

- the reason we need to set the env variables (the reason uv does not find the venv automatically) is probably because of the `cd ..`. From the [`uv run` docs](https://docs.astral.sh/uv/reference/cli/#uv-run):

  > When used outside a project, if a virtual environment can be found in the current directory or a parent directory, the command will be run in that environment. Otherwise, the command will be run in the environment of the discovered interpreter.

  so because of the `cd ..` the venv directory is actually in a directory _down_, and not in the current or parent directory.

- We need the `cd ..` in this case to have the tests run properly on the installed package, but in case we didn't and wanted to run `uv` from the root without considering it as a project, that might work by putting a `[tool.uv] managed = false` in the pyproject.toml